### PR TITLE
Unchangedtoast

### PR DESCRIPTION
**Pull request by mhagander**
I think this fixes the issue with unchanged toast datums, where decoder_raw previously generated invalid SQL.

AFAICS these can only occur in UPDATEs (otherwise they are new). And if they are unchanged, we can just skip them in the updates.

I'm not entirely certain there might not be other cases where there is an issue though, so I'd appreciate a review :)

And this time I even included regression tests.

### CODE REVIEW

**Comment by mhagander** [comment]
Hang on - don't apply. I think this might be causing crashes in a non-test system :)

**Comment by mhagander** [comment]
Was missing a null-check. I've force-pushed a version that includes the checks. So please review again :)

File: decoder_raw/expected/basic.out [hunk 1]
```diff
@@ -319,6 +319,23 @@ SELECT data FROM pg_logical_slot_get_changes('custom_slot', NULL, NULL, 'include
...
...xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx');
```
Comment (michaelpq) [inline decoder_raw/expected/basic.out]: I think I am going to use a substr() here...

**Comment by michaelpq** [comment]
Just did some adjustments, using substr(), and pushed as 4f30d5c.

**Comment by michaelpq** [comment]
Btw, I have added some safeguards to detect those unchanged toast values, and was not able to get some for DELETE and UDPDATE queries, particularly WHERE clauses even if REPLICA IDENTITY is FULL, so your fix looked fine to me.

**Comment by mhagander** [comment]
Fair enough -- I'd use the subsstr() towards the end of the string instead "just in case", since the whole point is testing the long data. Not that it's likely to show up truncated, but it never hurts...

**Comment by michaelpq** [comment]
Okay, I have changed the test so as it looks as well at the end of the string. That's a cheap insurance.

**Pull request closed without merging**